src/modules/profile.module.tsx

```diff
@@ -35,58 +27,47 @@ const useStyles = makeStyles(theme => ({
   }
 }))
 
-const resolveStatus = (state: ApplicationState): ProfileStatus => {
-  if (!state) {
-    return "start"
-  }
-
-  switch (state) {
-    case "DRAFT":
-      return "draft"
-    case "SUBMITTED":
-      return "submitted"
-    case "FILE_CHECKED":
-      return "fileChecked"
-    case "REJECTED_RESOLVE_FILE_ISSUE_TOO_LATE":
-      return "fileRejected"
-    case "INVITED_TO_INTERVIEW":
-      return "invitedToInterview"
-    case "REJECTED_NOT_INVITED_TO_INTERVIEW":
-      return "notInvitedToInterview"
-    case "PASSED_INTERVIEW":
-      return "passedInterview"
-    case "REJECTED_FAILED_THE_INTERVIEW":
-      return "failedInterview"
-    case "PAYMENT_ACCEPTED":
-      return "paymentAccepted"
-    default:
-      return "start"
-  }
-}
-
 const ProfileModule = () => {
   const classes = useStyles()
   const { me } = useAuthContext()
+  const { application } = useApplicationStateContext()
 
   const { lgCode, applicationState, firstname, lastname } = me.data as MyProfileModel
 
-  const profileStatus = resolveStatus(applicationState)
+  function resolveFileStatus(file1Status: FileStatus, file2Status: FileStatus, file3Status: FileStatus) {
+    const files = [file1Status, file2Status, file3Status]
+    if (files.includes("EMPTY")) return "EMPTY"
+    if (files.includes("CHANGE_REQUIRED")) return "CHANGE_REQUIRED"
+    if (files.includes("UPLOADED")) return "UPLOADED"
+    return "PASSED"
+  }
+
+  const fileStatus: FileStatus = resolveFileStatus(application.photo.status, application.parentalConsent.status, application.transcript.status)
+
+  const resolveProfileStatus = {
+    NOT_FILLED: "start" as ProfileStatus,
+    DRAFT: "draft" as ProfileStatus,
+    SUBMITTED: {
+      EMPTY: "draft" as ProfileStatus, // This state will not happen
+      UPLOADED: "submitted" as ProfileStatus,
+      CHANGE_REQUIRED: "fileRejected" as ProfileStatus,
+      PASSED: "fileChecked" as ProfileStatus // This state will not happen
+    },
+    FILE_CHECKED: "fileChecked" as ProfileStatus,
+    REJECTED_RESOLVE_FILE_ISSUE_TOO_LATE: "fileRejected" as ProfileStatus,
+    INVITED_TO_INTERVIEW: "invitedToInterview" as ProfileStatus,
+    REJECTED_NOT_INVITED_TO_INTERVIEW: "notInvitedToInterview" as ProfileStatus,
+    PASSED_INTERVIEW: "passedInterview" as ProfileStatus,
+    REJECTED_FAILED_THE_INTERVIEW: "failedInterview" as ProfileStatus,
+    PAYMENT_ACCEPTED: "paymentAccepted" as ProfileStatus
```

Comment:
you can use `Enum` instead for coding consistency and avoiding `as ProfileStatus `

Reply:
`as` is used for hacking typescript. Not recommend